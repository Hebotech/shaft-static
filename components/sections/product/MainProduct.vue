<template>
  <div>
    <client-only>
      <swiper :slides-per-view="3" :space-between="50">
        <swiper-slide>
          <img
            v-if="activeProduct"
            :src="`${activeProduct.images[counter]}`"
            class="main-product"
            loading="lazy"
        /></swiper-slide>
        <swiper-slide>
          <img
            v-if="activeProduct"
            :src="`${activeProduct.images[counter]}`"
            class="main-product"
            loading="lazy"
        /></swiper-slide>
        <swiper-slide>Slide 3</swiper-slide>
      </swiper>
    </client-only>
  </div>
</template>

<script>
import { productHelmetMixin } from '@/mixins/productHelmetsMixin';

export default {
  name: 'MainProduct',

  mixins: [productHelmetMixin],
};
</script>

<style lang="scss" scoped>
.main-product {
  cursor: pointer;
  background-position: center;
  background-size: cover;
  align-content: center;
  background-attachment: scroll;
  height: 100%;
  width: 35rem;
  filter: drop-shadow(4px 9.69px 20px rgba(0, 0, 0, 0.452));
}
@media (max-width: 576px) {
  .main-product {
    height: auto;
    width: 350px;
  }
  .circle {
    position: absolute;
    top: 10%;
    left: 15%;
    width: 0px;
    height: 0px;
    align-self: center;
    padding: 35%;
  }

  img {
    max-width: 40%;
  }
}
@media (max-width: 768px) {
  .main-product {
    width: 350px;
  }
  img {
    max-width: 90%;
  }
}
</style>
