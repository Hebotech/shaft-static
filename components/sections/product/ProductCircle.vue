<template>
  <div
    class="bg-circle row mb-3 mt-5 d-flex justify-content-between align-items-center order-0 order-lg-2 d-flex"
  >
    <non-active-products />

    <div class="col-8 justify-content-center">
      <main-product :counter="counter" />
    </div>
  </div>
</template>

<script>
import { productHelmetMixin } from '@/mixins/productHelmetsMixin';
import nonActiveProducts from './NonActiveProducts.vue';
import MainProduct from './MainProduct.vue';

export default {
  name: 'ProductCircle',
  components: {
    nonActiveProducts,
    MainProduct,
  },
  mixins: [productHelmetMixin],
  mounted() {
    window.addEventListener('keyup', (event) => {
      switch (event.keyCode) {
        case 37:
          this.lastImage();
          break;
        case 38:
          this.lastProduct();
          break;
        case 39:
          this.nextImage();
          break;
        case 40:
          this.nextProduct();
          break;
        // no default
      }
    });
  },
};
</script>

<style lang="scss" scoped></style>
