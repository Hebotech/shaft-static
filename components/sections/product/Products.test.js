import Vuex from 'vuex';
import {mount, shallowMount, createLocalVue} from '@vue/test-utils';
import MainProduct from './MainProduct.vue';
import ProductCircle from './ProductCircle.vue';
import productsMixin from '../../../mixins/productHelmetsMixin.js';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
  state:{
    jel:'asda',
  },
  mutations:{
    ADDING_COUNTER:()=>jest.fn(),
    RESTART_COUNTER:()=>jest.fn(),
  },
  actions:{
    lastProduct:()=> jest.fn(),
    nextProduct:()=> jest.fn(),
  }
});

const mixins = {
  productsMixin
};

const data = () =>Â {
    return {
      activeProduct:{
       images:['asdasda.png','superDuper.png']
      }
    }
}

describe('All products components from sections [sections/product]', ()=>{
  it('MainProduct', ()=>{
    let cmp = mount(MainProduct, {store, localVue, mixins});
    expect(cmp.html()).toMatchSnapshot();
  })

  it('ProductCircle trigger prodcucts (in mixin) with key up', () => {
    let cmp = shallowMount(ProductCircle, {store, localVue, data,mixins, attachToDocument:true});
    spyOn(cmp.vm, 'lastImage');

    window.dispatchEvent(new KeyboardEvent('keyup', { keyCode: 37 }))
    expect(cmp.vm.lastImage).toBeCalled();
  });

  it('ProductCircle trigger prodcucts (in mixin) with key down', () => {
    let cmp = shallowMount(ProductCircle, {store, localVue,data, mixins, attachToDocument:true});
    spyOn(cmp.vm, 'lastProduct');

    window.dispatchEvent(new KeyboardEvent('keyup', { keyCode: 38 }))
    expect(cmp.vm.lastProduct).toBeCalled();
  });

  it('ProductCircle trigger prodcucts (in mixin) with key right', () => {
    let cmp = shallowMount(ProductCircle, {store, localVue,data, mixins, attachToDocument:true});
    spyOn(cmp.vm, 'nextImage');''

    window.dispatchEvent(new KeyboardEvent('keyup', { keyCode: 39 }))
    expect(cmp.vm.nextImage).toBeCalled();
  });

  it('ProductCircle trigger nextProduct with key left', () => {
    let cmp = shallowMount(ProductCircle, {store, localVue,data, mixins, attachToDocument:true});
    spyOn(cmp.vm, 'nextProduct');

    window.dispatchEvent(new KeyboardEvent('keyup', { keyCode: 40 }))
    expect(cmp.vm.nextProduct).toBeCalled();
  });
});