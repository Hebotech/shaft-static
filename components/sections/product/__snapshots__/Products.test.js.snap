// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All products components from sections [sections/product] MainProduct 1`] = `
"<div>
  <!---->
  <div class=\\"circle\\"></div>
</div>"
`;

exports[`All products components from sections MainProduct 1`] = `
"<div>
  <!---->
  <div class=\\"circle\\"></div>
</div>"
`;
